core_data_modules/traced_data/util/fold_traced_data.py

```diff
@@ -70,6 +70,40 @@ def assert_equal_keys_equal(td_1, td_2, equal_keys):
                                                    "different (has values '{}' and '{}' " \
                                                    "respectively)".format(key, td_1.get(key), td_2.get(key))
 
+    @staticmethod
+    def reconcile_missing_values(value_1, value_2):
+        """
+        Reconciles two missing values, by choosing the form of missing value with the highest precedence.
+
+        The precedence order for missing values is defined as follows (highest precedence listed first):
+         - Codes.STOP
+         - Codes.NOT_REVIEWED
+         - Codes.NOT_CODE
+         - Codes.TRUE_MISSING
+         - Codes.SKIPPED
+         - Codes.NOT_LOGICAL
+         - None
+
+        :param value_1: Code to reconcile.
+        :type value_1: str
+        :param value_2: Code to reconcile.
+        :type value_2: str
+        :return: Reconciled code.
+        :rtype: str
+        """
+        # Precedence order in case of conflicts; highest precedence first
+        precedence_order = [
```

Comment:
Would it better to define this in Codes so it's obvious that if one gets added it needs to be added to the precedence order as well?

Reply:
If we do then there may be other references to missing codes which we should move into Codes as well. In this case, if a new missing code is seen the program will fail with an AssertionError...